monkey/infection_monkey/master/propagator.py

```diff
@@ -51,16 +51,30 @@ def _scan_network(self, propagation_config: Dict, stop: Event):
 
         logger.info("Finished network scan")
 
-    def _process_scan_results(
-        self, ip: str, ping_scan_data: PingScanData, port_scan_data: Dict[int, PortScanData]
-    ):
+    def _process_scan_results(self, ip: str, scan_results: IPScanResults):
         victim_host = VictimHost(ip)
-        has_open_port = False
 
+        Propagator._process_ping_scan_results(victim_host, scan_results.ping_scan_data)
+        has_open_port = Propagator._process_tcp_scan_results(
+            victim_host, scan_results.port_scan_data
+        )
```

Comment:
`_process_tcp_scan_results` and `has_open_port` should be separate methods. Result processing can happen in the Propagator, but `has_open_port()` should be part of port scan data interface.

Reply:
The `IPScanner` already had a private static method for this. I've just made it public for now. I'd like to change `PortScanData` to be more well-defined, but I haven't quite figured out a way that I'm happy with yet.